### PR TITLE
Introduce NodeLedger as an intermediate Ledger type

### PR DESCRIPTION
**Pull request by Geod24**
```
A NodeLedger is a more involved Ledger than the basic Ledger,
as the basic Ledger should be usable by a client without some of
the machinery present in the NodeLedger, most notably the two pools.

As a first step, and instant win to decouple things, we can move
`onValidatedBlock` to the `NodeLedger`, as a client would most likely
perform operations sequentially (call `acceptBlock` then check state).
```

Extracted from https://github.com/bosagora/agora/pull/2804

### CODE REVIEW

File: source/agora/consensus/Ledger.d [hunk 1]
```diff
@@ -1485,6 +1468,61 @@ public class Ledger
     }
 }
 
+/// The Ledger class held by a node
```
Comment (hewison-chris) [inline source/agora/consensus/Ledger.d:1471]: Due to the large size of  this file is it maybe better to create `NodeLedger.d` file?

File: source/agora/consensus/Ledger.d [hunk 2]
```diff
@@ -1516,7 +1554,7 @@ unittest
 
 *******************************************************************************/
 
-public class ValidatingLedger : Ledger
+public class ValidatingLedger : NodeLedger
```
Comment (hewison-chris) [inline source/agora/consensus/Ledger.d:1557]: Also maybe a new file. But maybe not part of this commit.

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/agora/consensus/Ledger.d (hunk 1 above)
Comment (Geod24) [reply source/agora/consensus/Ledger.d:1471]: Check the linked PR, it does that in one of the last commits, but first I need to break down the dependencies.

Comment (hewison-chris) [reply source/agora/consensus/Ledger.d:1471]: Great minds ...
